api_test.go

```diff
@@ -0,0 +1,230 @@
+			RowKeys:    rowKeys,
+			ColumnIDs:  colIDs,
+			Timestamps: timestamps,
+		}
+		if err := m0.API.Import(ctx, req); err != nil {
+			t.Fatal(err)
+		}
+
+		pql := fmt.Sprintf("Row(%s=%s)", field, rowKey)
+
+		// Query node0.
+		if res, err := m0.API.Query(ctx, &pilosa.QueryRequest{Index: index, Query: pql}); err != nil {
+			t.Fatal(err)
+		} else if columns := res.Results[0].(*pilosa.Row).Columns(); !reflect.DeepEqual(columns, colIDs) {
+			t.Fatalf("unexpected column ids: %+v", columns)
+		}
+
+		// Query node1.
+		if res, err := m1.API.Query(ctx, &pilosa.QueryRequest{Index: index, Query: pql}); err != nil {
+			t.Fatal(err)
+		} else if columns := res.Results[0].(*pilosa.Row).Columns(); !reflect.DeepEqual(columns, colIDs) {
+			t.Fatalf("unexpected column ids: %+v", columns)
+		}
+	})
+}
+func TestAPI_ImportValue(t *testing.T) {
+	c := test.MustRunCluster(t, 2,
+		[]server.CommandOption{
+			server.OptCommandServerOptions(
+				pilosa.OptServerNodeID("node0"),
+				pilosa.OptServerClusterHasher(&moddHasher{}),
+			)},
+		[]server.CommandOption{
+			server.OptCommandServerOptions(
+				pilosa.OptServerNodeID("node1"),
+				pilosa.OptServerClusterHasher(&moddHasher{}),
+			)},
+	)
+	defer c.Close()
+
+	m0 := c[0]
+	m1 := c[1]
+
+	t.Run("ValColumnKey", func(t *testing.T) {
+		ctx := context.Background()
+		index := "valck"
+		field := "f"
+
+		_, err := m0.API.CreateIndex(ctx, index, pilosa.IndexOptions{Keys: true})
+		if err != nil {
+			t.Fatalf("creating index: %v", err)
+		}
+		_, err = m0.API.CreateField(ctx, index, field, pilosa.OptFieldTypeInt(0, 100))
+		if err != nil {
+			t.Fatalf("creating field: %v", err)
+		}
+
+		// Generate some keyed records.
+		values := []int64{}
+		colKeys := []string{}
+		for i := 1; i <= 10; i++ {
+			values = append(values, int64(i))
+			colKeys = append(colKeys, fmt.Sprintf("col%d", i))
+		}
+
+		// Import data with keys to the coordinator (node0) and verify that it gets
+		// translated and forwarded to the owner of shard 0 (node1; because of moddHasher)
+		req := &pilosa.ImportValueRequest{
+			Index:      index,
+			Field:      field,
+			Shard:      0,
+			ColumnKeys: colKeys,
+			Values:     values,
+		}
+		if err := m0.API.ImportValue(ctx, req); err != nil {
+			t.Fatal(err)
+		}
+
+		pql := fmt.Sprintf("Range(%s>0)", field)
+
+		// Query node0.
+		if res, err := m0.API.Query(ctx, &pilosa.QueryRequest{Index: index, Query: pql}); err != nil {
+			t.Fatal(err)
+		} else if keys := res.Results[0].(*pilosa.Row).Keys; !reflect.DeepEqual(keys, colKeys) {
+			t.Fatalf("unexpected column keys: %+v", keys)
+		}
+
+		// Query node1.
+		if res, err := m1.API.Query(ctx, &pilosa.QueryRequest{Index: index, Query: pql}); err != nil {
+			t.Fatal(err)
+		} else if keys := res.Results[0].(*pilosa.Row).Keys; !reflect.DeepEqual(keys, colKeys) {
+			t.Fatalf("unexpected column keys: %+v", keys)
+		}
+	})
+}
+
+// moddHasher represents a simple, mod-based hashing offset by 1.
+type moddHasher struct{}
```

Comment:
It was a little bit of mod+odd since it's a mod hasher that's offset by 1. But I get that that may just be confusing. I may tweak that a little anyway, so I'll update this PR.